motor_control/throttle/button_control.ino

```diff
@@ -8,18 +8,70 @@ ButtonThrottle::ButtonThrottle(const int &pin, int timeToFull) : Throttle(pin, t
 
 int ButtonThrottle::mappingFunction(int time_pressed, m_map_type mapType)
 {
-  if (mapType == LINEAR)
+  if(m_isPressed == true)
   {
-    return std::min(time_pressed / m_timeToFull, 1) * 100;
+if (mapType == LINEAR)
+  {
+    m_speedOfCar = std::min((double)time_pressed / (double)m_timeToFull, 1.0) * m_maxValue;
+    return m_speedOfCar;
+  }
+  if (mapType == POWER)
+  {
+    if (time_pressed <= m_timeToFull)
+    {
+      m_speedOfCar = std::pow((double)time_pressed / (double)m_timeToFull, 2) * m_maxValue;
+      return m_speedOfCar; // This function uses m_timeToFull to create a dependency on what time we pick and creates an exponential to the power of the 2 with the ratio
+    }
+    else
+    {
+      m_speedOfCar = m_maxValue;
+      return m_speedOfCar;
+    }
   }
   if (mapType == EXPONENTIAL)
   {
+    if (time_pressed < m_timeToFull)
+    {
+      double temp_time_pressed = time_pressed;
+      double ratio = 0;
+      ratio = temp_time_pressed / 1000;
+      if (ratio == 0)
+      {
+        return 0;
+      }
+      if (std::exp(ratio) <= m_maxValue)
+      {
+        m_speedOfCar = std::exp(ratio);
+        return m_speedOfCar; //This function has no reliance on m_timeToFull, it simply follows an e exponential curve up til maximum value. Takes about 5.5 seconds to reach max speed(assuming 255 is the value we set)
+      }
+    }
+    else
+    {
+      m_speedOfCar = m_maxValue;
+      return m_speedOfCar;
+    }
+  }
   }
-  if (mapType == LOGARITHMIC)
+}
+int mappingFunctionDeceleration(int time_NotPressed)
```

Comment:
I think this function is supposed to be called from compute_motor_value() but its not there currently